### ISSUE TITLE
Rank Advancement

### ISSUE THREAD

**Issue by tsitu** [tsitu/MH-Tools#46]
@groupsky 
With the recent addition of 3 new Hunter's Titles, advancement.csv values are now out of date. If you know who created [this](https://docs.google.com/spreadsheets/d/17Ss1wGMWNJiXAKOW4dsWXAuomqTJ_IHqypagJbD6Ax0/edit#gid=0) spreadsheet we could ask him/her for help. 

For reference, the new wisdom table is shown below, as calculated by Tan YiKai.

![Wisdom Mapping](https://i.imgur.com/8oTKSgL.png "Wisdom Mapping")

**Comment by groupsky** [comment]
I have not idea who the author is, found it in the [wiki](http://mhwiki.hitgrab.com/wiki/index.php/Rank#External_links). 

I'm not sure what the wisdom represents. I'm guessing that each mice have some wisdom value, and the progress is calculated as mouse wisdom divided by rank wisdom. If that is true, we should be able to deduce the mice wisdom, given the table, and then calculate rank progress on the fly. That would ease populating data for new mice and ranks.

@tsitu, do you have link to the table, as entering numbers from the image could introduce errors.

I'm working on MoPi, and I will not look into this task next few days (maybe weeks).

**Comment by tsitu** [comment]
Turns out it was [YiKai himself](http://www.mousehuntgame.com/forum/showthread.php?37328) who created the data collection spreadsheet over 6 years ago.

You're right that each mouse has a wisdom value. The formula should be as follows:
`Advancement % = Mouse Wisdom / (Next Rank Wisdom - Current Rank Wisdom)`

Since we have both next rank and current rank wisdom from the image (which are integers and likely accurate - I suspect YiKai pulls data directly from the MH Android APK), all we need to migrate to the new rank scaling system is each mouse's wisdom value. For new mice not in the spreadsheet, that would simply be the difference in wisdom before and after a hunt, which can be tracked [here](http://www.mousehuntgame.com/item.php?item_type=wisdom_stat_item). Note that there is a base passive hunt (aka trap check) wisdom of 15 and an active hunt wisdom of 25.

_Example 1: White Mouse, calculate Adv% for Initiate -> Journeyman_ 
Using 1.75% from Cell C5: Mouse Wisdom = 0.0175 * (2000 - 0) = 35
Using (old) wisdom values from above: Adv% = 35 / (34276 - 18250) = 0.00218395108 or 0.218% in Cell F5
Using (new) wisdom values from above: Adv% = 35 / (28477 - 12944) = 0.00225326724 or 0.225%

_Example 2: Cycloness Mouse, calculate Adv% for Archduke -> Viceroy_
Not in spreadsheet, derived Mouse Wisdom from mousehuntgame.com: 7525
7525 - 25 (base active hunt wisdom) = 7500
Using (new) wisdom values from above: Adv% = 7500 / (366069255 - 166395116) = 0.00003756119 or 0.00375611986% which appears accurate!

**TL;DR** Since you're hard at work on MoPi I'll try to calculate all the individual mouse wisdom values in a new sheet. The spreadsheet has 824/910 breeds so 86 still have to be done manually.

**Comment by tsitu** [comment]
YiKai's updated wisdom mapping using user.title_wisdom and MH's API

![Wisdom Mapping](https://i.imgur.com/vFAKoa6.png "Wisdom Mapping")
